### DISCUSSION TITLE
How to add short wave (solar) radiation at sea surface?

### DISCUSSION THREAD

**Discussion by jhdong2016** [CliMA/Oceananigans.jl#3218] [Computational science!] [answered]
Hi, I am using Oceananigans to simulate the ocean surface boundary layer. It is easy to add net sea surface buoyancy flux. However, I can not find any introduction to how to add solar radiation separately. Is there anyone that knows the method?

**Comment by glwagner** [comment]
Would you like to add a radiative flux just at the surface? In that case, it's specified the same way as any flux.

If you'd like to add an interior flux, then you can specify the flux divergence as an interior `Forcing`.

If you can add some specifics to your question (perhaps with some code) we can help provide a more specific answer!

**Reply by jhdong2016** [reply]
The solar radiation can penetrate the ocean interior. So, I believe this should be a diminishing flux from the surface to the interior. Is there any suggestion to realize it? The following code is a case in which I use a net flux for the ocean to get buoyancy (i.e., no solar radiation). Right now, I want to make it closer to the real ocean, and want to add solar radiation. Also, the whole code is attached. 

=========================================================
```
# surface wind flux
u₁₀ = 3    # m s⁻¹, average wind velocity 10 meters above the ocean
cᴰ = 2.5e-3 # dimensionless drag coefficient
ρₐ = 1.225  # kg m⁻³, average density of air at sea-level
ρₒ = 1026.0 # kg m⁻³

Qᵘ = - ρₐ / ρₒ * cᴰ * u₁₀ * abs(u₁₀) # m² s⁻²

v_boundary_conditions = FieldBoundaryConditions(top = FluxBoundaryCondition(Qᵘ))

# surface buoyanc flux (or heat flux)
Qᵇ = 1e-8 # m² s⁻³, surface buoyancy flux
N² = 2e-5 # s⁻², initial and bottom buoyancy gradient

b_boundary_conditions = FieldBoundaryConditions(top = FluxBoundaryCondition(Qᵇ),
                                                bottom = GradientBoundaryCondition(N²))
coriolis = FPlane(f=1e-4) # s⁻¹
```


===========================================================
[si_langmuir_case0.txt](https://github.com/CliMA/Oceananigans.jl/files/12416933/si_langmuir_case0.txt)

**Reply by jagoosw** [reply]
In this case is the water attenuation constant/pre-calculable? If so you could make a body forcing that is just a function of z times the surface irradiance? 

If not (i.e. if the attenuation is dependant on temperature or the concentration of something) then you would probably have to have a separate field that you integrate from the surface at each time step like we do here: https://github.com/OceanBioME/OceanBioME.jl/blob/main/src/Light/2band.jl

**Reply by jhdong2016** [reply]
I believe that I want to use a constant attenuation. Could you please show me an example of how to use a body forcing?

**Reply by jagoosw** [reply, marked as answer]
In which case you could do something like:

```
heating(x, y, z, t, parameters) = parameters.surface_flux * exp(params.attenuation * z)

T_forcing = Forcing(heating, parameters = (surface_flux = 100, attenuation = 0.1))

...

model = NonhydrostaticModel(..., forcing = (T = T_forcing, ))

```

**Reply by jhdong2016** [reply]
That's great! I will try this in my case. Really thanks for the help!

**Reply by glwagner** [reply]
Here's the docs on `Forcing` in case it's helpful:

https://clima.github.io/OceananigansDocumentation/stable/model_setup/forcing_functions/
